payment/manual/manual.go

```diff
@@ -60,36 +67,61 @@ func ManualPaymentFx(w http.ResponseWriter, r *http.Request) (string, interface{
 
 	methods := models.GetAvailableMethods(authToken.Role)
 	if len(methods) == 0 {
-		return "", nil, fmt.Errorf("no methods available for manual payment")
+		err = fmt.Errorf("no methods available for manual payment")
+		return "", nil, err
 	}
 
 	isMethodAllowed := lib.SliceContains[string](methods, payload.PaymentMethod)
 	if !isMethodAllowed {
-		log.Printf("ERROR %s", errPaymentMethodNotAllowed)
-		return "", nil, fmt.Errorf(errPaymentMethodNotAllowed)
+		err = fmt.Errorf("ERROR %s", errPaymentMethodNotAllowed)
+		return "", nil, err
 	}
 
 	origin := r.Header.Get("Origin")
 	transactionUid := chi.URLParam(r, "transactionUid")
-	fireTransactions := lib.GetDatasetByEnv(origin, models.TransactionsCollection)
-	firePolicies := lib.GetDatasetByEnv(origin, models.PolicyCollection)
 
-	docsnap, err := lib.GetFirestoreErr(fireTransactions, transactionUid)
+	docsnap, err := lib.GetFirestoreErr(lib.TransactionsCollection, transactionUid)
 	if err != nil {
-		log.Printf("ERROR get transaction from firestore: %s", err.Error())
 		return "{}", nil, err
 	}
 	err = docsnap.DataTo(&transaction)
-	lib.CheckError(err)
+	if err != nil {
+		return "", nil, err
+	}
+
+	docsnap, err = lib.GetFirestoreErr(lib.PolicyCollection, transaction.PolicyUid)
+	if err != nil {
+		return "", nil, err
+	}
+	err = docsnap.DataTo(&policy)
+	if err != nil {
+		return "", nil, err
+	}
+
+	networkNode = network.GetNetworkNodeByUid(policy.ProducerUid)
+	if networkNode == nil {
+		err = errors.New("networkNode not found")
+		return "", nil, err
+	}
+	warrant = networkNode.GetWarrant()
+	ccAddress = mail.GetNetworkNodeEmail(networkNode)
+	flowName, _ = policy.GetFlow(networkNode, warrant)
+	log.Printf("flowName '%s'", flowName)
+
+	isOnlineProducer := authToken.UserID != policy.ProducerUid || flowName != models.RemittanceMgaFlow
+	if authToken.IsNetworkNode && isOnlineProducer {
+		err = errors.New("cannot access to transaction")
+		return "", nil, err
+	}
```

Comment:
I know that this might be a little obnoxious, but what if we controlled like this?
```golang
canUserAccessTransaction := authToken.Role == models.UserRoleAdmin || (authToken.IsNetworkNode && authToken.UserID == policy.ProducerUid && flowName == models.RemittanceMgaFlow)
if !canUserAccessTransaction {
        err = errors.New("user cannot access transaction")
        return "", nil, err
}
```